python3/extensions/pool_update.apply

```diff
@@ -153,10 +149,13 @@ if __name__ == '__main__':
                 shutil.rmtree(os.path.dirname(yum_conf_file))
             except Exception as e:
                 pass
+
     finally:
         if lock_acquired:
             lock.release()
-            if os.path.isfile(lock_file):
-                os.remove(lock_file)
+        try:
+            os.remove(lock_file)
+        except Exception as e:
+            pass
```

Comment:
This changes the context in which the `lock_file` is removed from `if lock_acquired:` to `finally:` (always, not just when the lock was actually acquired).

This means that if this would be merged, the `lock_file` would always be removed, even when `lock_file.acquire(blocking=False)` fails.

In turn, this means that the 3rd try (after the 2nd try removing the `lock_file`) to acquire the lock would always successfully "get" the lock, even if the 1st process is still inside the critical section.

Therefore, you must keep the context for removing the lock to be `if lock_acquired:`.

Hence, in other words: You must keep the indent level of the removal of the lock intact.

But, replacing this construct with `os.remove(lock_file)` (with Exceptions suppressed) is a good idea:
```py
            if os.path.isfile(lock_file):
                os.remove(lock_file)
```
This is better as it avoids the race of the `lock_file` disappearing (being deleted by another process) between `if os.path.isfile(lock_file):` and `os.remove(lock_file)`. This avoids the race, but you still have to keep it at the correct indent level.

With Python3, the most pythonic way of suppressing an exception is using contextlib.suppress(Exception) instead (of course, move `import contextlib` to the top and apply isort to it):
```suggestion
            import contextlib
            with contextlib.suppress(OSError):
                os.remove(lock_file)
```